subsys/bluetooth/mesh/shell.c

```diff
@@ -3335,227 +1215,146 @@ static int cmd_cdb_app_key_del(const struct shell *shell, size_t argc,
+#endif
+	SHELL_SUBCMD_SET_END);
 #endif
 
-	/* Health Server Model Operations */
-	SHELL_CMD_ARG(add-fault, NULL, "<Fault ID>", cmd_add_fault, 2, 0),
-	SHELL_CMD_ARG(del-fault, NULL, "[Fault ID]", cmd_del_fault, 1, 1),
+#if defined(CONFIG_BT_MESH_LOW_POWER)
+SHELL_STATIC_SUBCMD_SET_CREATE(low_pwr_cmds,
+	SHELL_CMD_ARG(lpn, NULL, "<Val: off, on>", cmd_lpn, 2, 0),
+	SHELL_CMD_ARG(poll, NULL, NULL, cmd_poll, 1, 0),
+	SHELL_SUBCMD_SET_END);
+#endif
 
-#if defined(CONFIG_BT_MESH_CDB)
-	/* Mesh Configuration Database Operations */
-	SHELL_CMD_ARG(cdb-create, NULL, "[NetKey]", cmd_cdb_create, 1, 1),
-	SHELL_CMD_ARG(cdb-clear, NULL, NULL, cmd_cdb_clear, 1, 0),
-	SHELL_CMD_ARG(cdb-show, NULL, NULL, cmd_cdb_show, 1, 0),
-	SHELL_CMD_ARG(cdb-node-add, NULL, "<UUID> <addr> <num-elem> "
-		      "<NetKeyIdx> [DevKey]", cmd_cdb_node_add, 5, 1),
-	SHELL_CMD_ARG(cdb-node-del, NULL, "<addr>", cmd_cdb_node_del, 2, 0),
-	SHELL_CMD_ARG(cdb-subnet-add, NULL, "<NeyKeyIdx> [<NetKey>]",
-		      cmd_cdb_subnet_add, 2, 1),
-	SHELL_CMD_ARG(cdb-subnet-del, NULL, "<NetKeyIdx>", cmd_cdb_subnet_del,
-		      2, 0),
-	SHELL_CMD_ARG(cdb-app-key-add, NULL, "<NetKeyIdx> <AppKeyIdx> "
-		      "[<AppKey>]", cmd_cdb_app_key_add, 3, 1),
-	SHELL_CMD_ARG(cdb-app-key-del, NULL, "<AppKeyIdx>", cmd_cdb_app_key_del,
-		      2, 0),
+/* List of Mesh subcommands.
+ *
+ * Each command is documented in doc/reference/bluetooth/mesh/shell.rst.
+ *
+ * Please keep the documentation up to date by adding any new commands to the
+ * list.
+ */
+SHELL_SUBCMD_SET_CREATE(mesh_cmds, (mesh));
+
+/* General operations */
+SHELL_SUBCMD_ADD((mesh), init, NULL, NULL, cmd_init, 1, 0);
+SHELL_SUBCMD_ADD((mesh), reset_local, NULL, NULL, cmd_reset, 1, 0);
+#if defined(CONFIG_BT_MESH_LOW_POWER)
+SHELL_SUBCMD_ADD((mesh), low_power, &low_pwr_cmds, "Low power Commands", shell_model_cmds_help, 0,
+		 0);
 #endif
 
-	SHELL_SUBCMD_SET_END
-);
+#if defined(CONFIG_BT_MESH_CDB)
+SHELL_SUBCMD_ADD((mesh), cdb, &cdb_cmds, "Configuration Database", shell_model_cmds_help, 0, 0);
+#endif
+#if defined(CONFIG_BT_MESH_GATT_PROXY) || defined(CONFIG_BT_MESH_PROXY_CLIENT)
+SHELL_SUBCMD_ADD((mesh), proxy, &proxy_cmds, "Proxy commands", shell_model_cmds_help, 0, 0);
+#endif
 
-static int cmd_mesh(const struct shell *shell, size_t argc, char **argv)
-{
-	if (argc == 1) {
-		shell_help(shell);
-		/* shell returns 1 when help is printed */
-		return 1;
-	}
+SHELL_SUBCMD_ADD((mesh), provisioning, &prov_cmds, "Provisioning commands", shell_model_cmds_help,
+		 0, 0);
 
-	shell_error(shell, "%s unknown parameter: %s", argv[0], argv[1]);
+SHELL_SUBCMD_ADD((mesh), test, &test_cmds, "Test commands", shell_model_cmds_help, 0, 0);
 
-	return -EINVAL;
-}
+/* Placeholder for model shell modules that is configured in the application */
+SHELL_SUBCMD_SET_CREATE(model_cmds, (models));
+SHELL_SUBCMD_ADD((mesh), models, &model_cmds, "Model Commands", shell_model_cmds_help, 0,
```

Comment:
I thought about it for a while. If the only supported models is the cfg cli and health cli the `models` subfolder would be a little bit unnecessary, but if we want to include our DS model support under the same shell instance I think  it would look neater if we had a subfolder for it. Otherwise the root cmd folder of `mesh` would potentially be cluttered again, which is what i wanted to avoid in the first place. It of course possible to keep the `models` subfolder anyhow, but the purpose of it would not be obvious for people exclusively using Zephyr.

Reply:
> Otherwise, I guess `mesh_cmds` can be initialized with `SHELL_STATIC_SUBCMD_SET_CREATE`?

Is there a huge benefit of doing so? Retaining the possibility to add commands directly to `mesh` from other modules could come in handy.

Reply:
> I thought about it for a while. If the only supported models is the cfg cli and health cli the models subfolder would be a little bit unnecessary, but if we want to include our DS model support under the same shell instance I think it would look neater if we had a subfolder for it. Otherwise the root cmd folder of mesh would potentially be cluttered again, which is what i wanted to avoid in the first place. It is of course possible to keep the models subfolder anyway, but the purpose of it would not be obvious for people exclusively using Zephyr.

Makes sense.

> Is there a huge benefit of doing so? Retaining the possibility to add commands directly to mesh from other modules could come in handy.

Yes, because commands added to `SHELL_STATIC_SUBCMD_SET_CREATE` can have dash in the name.